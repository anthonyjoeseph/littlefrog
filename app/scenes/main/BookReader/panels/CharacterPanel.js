// app/BookReader/LanguagePanel.js

import React, { Component } from 'react';
import {
  StyleSheet
} from 'react-native';

import SubPanel from './SubPanel';
import RadioButtonRow from './RadioButtonRow';

class CharacterPanel extends Component{
  constructor(props){
    super(props);

    this.characters = [
      "hum",
      "ali",
      "squ",
      "dar"
    ];
  }

  slide(){
    this.refs.subPanel.slide();
  }

  render(){
      return (
        <SubPanel
          ref='subPanel'
          style={[
            this.props.style,
            {
              flexDirection: 'row',
              justifyContent: 'center',
              alignItems: 'stretch'
            }]}>
            <RadioButtonRow
              options={this.characters}
              onSelection={
                function(selection){
                  this.props.onChangeCharacter(selection);
                }.bind(this)
              }
            />
        </SubPanel>
      );
  }
}

export default CharacterPanel;
