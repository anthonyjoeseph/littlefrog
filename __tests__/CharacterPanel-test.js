// __tests__/CharacterPanel-test.js
import 'react-native';
import React from 'react';
import CharacterPanel from '../app/scenes/main/BookReader/panels/CharacterPanel';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <CharacterPanel
      style={{
        bottom: 75,
        position: 'absolute',
        zIndex: 1,
        width: 400,
        left: 0,
        height: 100
      }}
      onChangeCharacter={ (character) => { } }
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
