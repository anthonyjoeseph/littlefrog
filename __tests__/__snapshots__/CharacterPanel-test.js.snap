exports[`test renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "stretch",
      "bottom": 75,
      "flexDirection": "row",
      "height": 100,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "transform": Array [
        Object {
          "translateX": -1000,
        },
      ],
      "width": 400,
      "zIndex": 1,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "yellow",
          },
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }>
        hum
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "white",
          },
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={Object {}}>
        ali
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "white",
          },
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={Object {}}>
        squ
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "white",
          },
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={Object {}}>
        dar
      </Text>
    </View>
  </View>
</View>
`;
